staticaddr/deposit/deposit.go

```diff
@@ -59,47 +59,26 @@ type Deposit struct {
 	sync.Mutex
 }
 
-// IsInPendingState returns true if the deposit is pending.
-func (d *Deposit) IsInPendingState() bool {
-	d.Lock()
-	defer d.Unlock()
-
-	return !d.IsInFinalState()
-}
-
 // IsInFinalState returns true if the deposit is final.
 func (d *Deposit) IsInFinalState() bool {
-	d.Lock()
-	defer d.Unlock()
-
-	return d.state == Expired || d.state == Withdrawn || d.state == Failed
+	return d.state == Expired || d.state == Withdrawn ||
```

Comment:
Good catch! I've removed the locks while working on the `recover()` in the withdrawal manager. There in `GetActiveDepositsInState` we lock the states of several deposits in `lockDeposits(deposits)`, but then try to lock again in check `if !d.IsInState(stateFilter)`.

I've restored the locks here, and also provided a method `d.IsInStateNoLock(stateFilter)` that I call in `GetActiveDepositsInState` in `recover` of the withdrawal manager, since all deposits are already locked.